### ISSUE TITLE
Paper?

### ISSUE THREAD

**Issue by Ironholds** [Ironholds/triebeard#8]
@wrathematics and @richfitz what would you think of doing a paper for the JSS on triebeard, ring and decker? "New data structures in the R programming language", going through:

1. The importance in performance and user convenience terms of picking the right DS for the job;
2. For each package, a description of the (theoretical) data structure and the (practical) usage;
3. Isn't it nice we have more options now, yes, that's very nice, look, they're used in packages and everything

**Comment by wrathematics** [comment]
There's an R Journal paper on some approaches to this already https://journal.r-project.org/archive/2015-1/oneil.pdf

But iirc, all of their data structures (including some more not mentioned in the paper, like Winston Chang's qstack) live on top of environments. So basically they're just sugar for hash tables. Whereas all of ours are legitimately new implementations, with various pros and cons (some examples here https://github.com/wrathematics/queue_benchmarking).

I'm interested in collaborating on something though.

**Comment by Ironholds** [comment]
Yeah, and we can always point to that paper and highlight the distinction for CYA purposes

**Comment by wrathematics** [comment]
Oh also, Jeff Horner's improved hash table for R would have positive performance impacts on the other approaches. Worth considering.

**Comment by richfitz** [comment]
I'm happy to be involved and help out, though I left academia to get away from papers 😉

My motivation for ring was pretty selfish though (https://github.com/richfitz/dde), but if it's useful for others, that's cool too.

**Comment by wrathematics** [comment]
I'm trying to escape academia. How did you get out? You must tell me your secrets.

**Comment by Ironholds** [comment]
Mate, you work at a university.

**Comment by Ironholds** [comment]
Also working on a B+-tree implementation, not sure if we'd want to include that.
